### PR TITLE
Suggest metrics

### PR DESCRIPTION
**Pull request by dgl**
Adds support for suggesting `__name__` so autocomplete should work.

### CODE REVIEW

File: pkg/store/store.go [hunk 1]
```diff
@@ -104,38 +104,51 @@ func (store *OpenTSDBStore) Series(
...
 }
 
 func (store *OpenTSDBStore) LabelValues(
 	ctx context.Context,
 	req *storepb.LabelValuesRequest) (*storepb.LabelValuesResponse, error) {
 	level.Debug(store.logger).Log("msg", "LabelValues", "Label", req.Label)
+	if req.Label == "__name__" {
+		var pNames []string
+		for _, item := range store.metricNames {
+			pNames = append(pNames, strings.Replace(item, ".", ":", -1))
```
Comment (szalai1) [inline pkg/store/store.go]: it would be worth putting it behind a feature flag since this is very CPU and memory-intensive.

Comment (dgl) [reply pkg/store/store.go]: Makes sense, done.

Comment (eswdd) [reply pkg/store/store.go]: why not just store the list of metric names twice? and then we don't have to do this replacement all the time but just one off every time we load the metrics?

Comment (eswdd) [reply pkg/store/store.go]: ah i see, you're loading the metric names every time for this, that feels.. unnecessary.. excessive?

Comment (dgl) [reply pkg/store/store.go]: It's not actually loading it, just copying from somewhere in memory. If I was going to optimise this then actually overriding Marshal on the protobuf and caching there would be the way to go (see how https://github.com/grpc/grpc-go/blob/master/encoding/proto/proto.go#L68 allows you to have your own proto.Marshaler), anything else feels a bit pointless as it'd waste memory but still not be optimal... I'll file an issue to look into optimising it later.

Comment (eswdd) [reply pkg/store/store.go]: ah yes, i see now the code has overtaken my comment.

**Pull request merged**